lib/signaling/v2/peerconnection.js

```diff
@@ -426,7 +430,9 @@ class PeerConnectionV2 extends StateMachine {
     if (!event.candidate) {
       peerConnectionState.ice.complete = true;
     }
-    this.emit('candidates', peerConnectionState);
+    if (!(this._isIceLite && event.candidate)) {
+      this.emit('candidates', peerConnectionState);
+    }
```

Comment:
This is the simplest way to emit the end-of-candidates event (and no ice candidates) after encountering an ice-lite agent. The approach I took was to emit the end of candidates event as soon the Client:

1. Applies a remote ice-lite answer.
2. Applies a local answer to an ice-lite offer.

The benefit is the signal comes sooner and more reliably (since gathering takes up to 40 seconds in Chrome). The drawback is more complexity in event handling. I'll let you decide what makes sense, because what you have here still meets the spirit of the RSP's [PeerConnection Handling](
https://code.hq.twilio.com/client/room-signaling-protocol/blob/master/docs/peerconnection_handling.md#ice-lite).

Reply:
@ceaglest ,

I think sending end of candidates when the PeerConnection actually finishes gathering candidates will be more accurate just from a logging perspective, unless we are gaining anything performance-wise by sending it sooner.

Reply:
I am okay either way, but I would also like to report ice gathering to insights in the future 👍.